client/src/components/home/index.js

```diff
@@ -156,10 +170,28 @@ function Home() {
         *  User is logged in, editing url details
         *  try updating the url details and handle the errors*/
       try {
+        const newDetails = {
+          title: urlTitle,
+          description: urlDescription,
+          shortUrl: updatedShortUrl,
+        };
+        /**  Check if details have been updated or not
+          *  If yes then proceed to send request to server
+          *  Otherwise display appropriate message to the user
+        */
+        if (JSON.stringify(newDetails) === JSON.stringify(details)) {
```

Comment:
Do not use `JSON.stringify` function for this. It can cause unintended bugs.
![image](https://user-images.githubusercontent.com/29327152/97111231-b7beaf00-1703-11eb-9676-81eaef8915a4.png)
In the above image, you may expect `a` to be equal to `b`. But it's not.
Since we have exactly 3 keys in the details object, comparing them individually would make sense.